ocaml/xapi/repository.ml

```diff
@@ -114,3 +129,476 @@ let cleanup_pool_repo () =
+  match Sys.file_exists repo_dir with
+  | true ->
+    (try
+       let cachedir = get_cachedir !Xapi_globs.pool_repo_name in
+       let md = parse_repomd (Filename.concat cachedir "repomd.xml") in
+       let updateinfo_xml_gz_path =
+         Filename.concat repo_dir (md.checksum ^ "-updateinfo.xml.gz")
+       in
+       ignore (Helpers.call_script !Xapi_globs.createrepo_cmd [repo_dir]);
+       begin match Sys.file_exists updateinfo_xml_gz_path with
+         | true ->
+           with_updateinfo_xml updateinfo_xml_gz_path (fun xml_file_path ->
+               let repodata_dir = Filename.concat repo_dir "repodata" in
+               ignore (Helpers.call_script !Xapi_globs.modifyrepo_cmd
+                         ["--remove"; "updateinfo"; repodata_dir]);
+               ignore (Helpers.call_script !Xapi_globs.modifyrepo_cmd
+                         ["--mdtype"; "updateinfo"; xml_file_path; repodata_dir]));
+           with_pool_repository (fun () ->
+               set_available_updates ~__context ~self)
+         | false ->
+           error "No updateinfo.xml.gz found: %s" updateinfo_xml_gz_path;
+           raise Api_errors.(Server_error (invalid_updateinfo_xml, []))
+       end
+     with
+     | Api_errors.(Server_error (code, _)) as e when code <> Api_errors.internal_error ->
+       raise e
+     | e ->
+       error "Creating local pool repository failed: %s" (ExnHelper.string_of_exn e);
+       raise Api_errors.(Server_error (createrepo_failed, [])))
+  | false ->
+    error "local pool repository directory '%s' does not exist" repo_dir;
+    raise Api_errors.(Server_error (reposync_failed, []))
+
+let with_local_repository ~__context f =
+  let master_addr =
+    try Pool_role.get_master_address ()
+    with Pool_role.This_host_is_a_master ->
+      Option.get (Helpers.get_management_ip_addr ~__context)
+  in
+  let url = Printf.sprintf "https://%s/repository/" master_addr in
+  Xapi_stdext_pervasives.Pervasiveext.finally
+    (fun () ->
+      remove_repo_conf_file !Xapi_globs.local_repo_name;
+      write_yum_config url !Xapi_globs.local_repo_name;
+      let config_params =
+          [
+            "--save";
+            Printf.sprintf "--setopt=%s.sslverify=false" !Xapi_globs.local_repo_name;
+            !Xapi_globs.local_repo_name
+          ]
+      in
+      ignore (Helpers.call_script !Xapi_globs.yum_config_manager_cmd config_params);
+      f ())
+    (fun () -> remove_repo_conf_file !Xapi_globs.local_repo_name)
+
+let assert_yum_error output =
+  let errors = ["Updateinfo file is not valid XML"] in
+  let open Xapi_stdext_std.Xstringext in
+  List.iter (fun err ->
+    if String.has_substr output err then (
+      error "Error from 'yum updateinfo list': %s" err;
+      raise Api_errors.(Server_error (invalid_updateinfo_xml, [])))
+    else ()) errors;
+  output
+
+let pkg_of_fullname s =
+  (* s I.E. "libpath-utils-0.2.1-29.el7.x86_64" *)
+  let r = Re.Str.regexp "^\\(.+\\)\\.\\(noarch\\|x86_64\\)$" in
```

Comment:
You can use `{| |}` and avoid the double escape :)

```
{|^\(.+\)\.\(noarch\|x86_64\)$|}
```

Reply:
Indeed it would be much more clear to get rid of these backslashes. But not got this passed build after trying with long time :(

Reply:
Okay, nevermind! It is clear what's going on right now anyway